### PR TITLE
First chunk of code for cmd line options

### PR DESCRIPTION
**Pull request by pcoop**
More to follow... but I've just seen the requirement for signing commits is about to go live and I have several commits in-flight... so figured I should try to get this up sooner rather than later.

### CODE REVIEW

**Review by matthewpwilson — CHANGES_REQUESTED** [review]
I think the error messages "Incompatible security credentials exist ..." are a bit cryptic. Suggest something that's about more specific about what doesn't "Different passwords were specified for the same user ID in the x and y parameter"

**Comment by pcoop** [comment]
One random observation I've discovered: Zowe's cics profile has a default port value of 1490, whereas the zowe cics-plugin has a default port of 449. Odd. Worse, the command line value clashes with the default zoSMF port. I've changed to use 1490 as the default for our code.

File: src/api/BundlePush/CmciConfig.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    if (params.arguments.cru !== undefined) {
+      cmciProfile.rejectUnauthorized = params.arguments.cru;
+      cmciProfile.ru = params.arguments.cru;
+    }
+
+    if (params.arguments.cpr !== undefined) {
+      cmciProfile.protocol = params.arguments.cpr;
+      cmciProfile.u = params.arguments.cpr;
```
Comment (matthewpwilson) [inline src/api/BundlePush/CmciConfig.ts]: I believe this should be `cmciProfile.o`

File: src/api/BundlePush/ZosmfConfig.ts [hunk 2]
```diff
@@ -62,22 +62,28 @@ export class ZosmfConfig {
...
+
+    // And for reject-unauthorized
     if (zosmfProfile.rejectUnauthorized === undefined) {
-      throw new Error("Required parameter --zosmf-reject-unauthorized is not set.");
+       zosmfProfile.rejectUnauthorized = true;
+       zosmfProfile.ru = true;
     }
   }
 
+  private static checkValueFound(value: string, parm: string) {
```
Comment (matthewpwilson) [inline src/api/BundlePush/ZosmfConfig.ts:84]: Could this function be pushed into a superclass rather than duplicated in `SshConfig` and `ZosmfConfig`?

**Review by matthewpwilson — CHANGES_REQUESTED** [review]

File: src/api/BundlePush/CmciConfig.ts (hunk 1 above)
Comment (pcoop) [reply src/api/BundlePush/CmciConfig.ts]: Ah yes, well spotted.

File: src/api/BundlePush/ZosmfConfig.ts (hunk 2 above)
Comment (pcoop) [reply src/api/BundlePush/ZosmfConfig.ts:84]: It could be... hardly seems worth the effort though.

**Review by matthewpwilson — APPROVED** [review]

**Pull request merged**